tests/UrlGeneratorTest.php

```diff
@@ -183,4 +193,120 @@ public function testDefaultNotUsedForRequiredParameter(): void
+    }
+
+    public function testAbsoluteUrlFromMethodHostParamWithTrailingSlashGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index', [], null, 'http://mysite.com/');
+
+        $this->assertEquals('http://mysite.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromRouteHostParamGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromRouteHostParamWithSchemeParamGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index', [], 'https');
+
+        $this->assertEquals('https://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromRouteHostParamWithTrailingSlashGenerated(): void
+    {
+        $routes = [
+            Route::get('/home/index')->name('index')->host('http://test.com/'),
+        ];
+        $url = $this->createUrlGenerator($routes)->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromCurrentHostGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index'),
+        ];
+
+        $router = $this->createUrlGenerator($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromCurrentHostWithNonStandardPortGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com:8080/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index'),
+        ];
+
+        $router = $this->createRouter($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com:8080/home/index', $url);
+    }
+
+    public function testAbsoluteUrlFromUrlHostParamWithProtocolRelativeSchemeGenerated(): void
+    {
+        $request = new ServerRequest('GET', 'http://test.com/home/index');
+        $routes = [
+            Route::get('/home/index')->name('index')->host('//test.com'),
+        ];
+
+        $router = $this->createRouter($routes);
+        $router->match($request);
+        $url = $router->generateAbsolute('index');
+
+        $this->assertEquals('http://test.com/home/index', $url);
```

Comment:
That doesn't look correct to me:

1. Host protocol is specified explicitly as `//`.
2. There's no override in `generateAbsolute()`.

So the result should be `//test.com/home/index`.